R/zchunk_LA114.wind-energy.R

```diff
@@ -28,43 +30,241 @@ module_energy_LA114.wind_DISABLED <- function(command, ...) {
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     Zhou_wind_supply_ctry_EJ <- get_data(all_data, "energy/Zhou_wind_supply_ctry_EJ")
-    A14.wind_curves <- get_data(all_data, "energy/A14.wind_curves")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # Silence package notes
+    `mid-price` <- `curve-exponent` <- `base-price` <- base.price <- iso <- GCAM_region_ID <- curve.exponent <-
+      max.price <- maxSubResource <- p <- supply <- optimize <- resource <- subresource <- maxSubResource <-
+      optimize <- mid.price <- . <- price <- NULL
+
+    # First, define three useful functions that will be used later: evaluate_smooth_res_curve,
+         # smooth_res_curve_approx_error, and generate_max_prices.
+
+    # The function, evaluate_smooth_res_curve, computes the smooth renewable resource function
+    # supply = (p - base.price) ^ curve.exponent / (mid.price ^ curve.exponent +
+    #    (p - base.price) ^ curve.exponent * maxSubResource
+    evaluate_smooth_res_curve <- function(curve.exponent, mid.price, base.price, maxSubResource, p) {
```

Comment:
In LA115.RooftopPV (GCAM-USA), there are similar but not exactly the same functions. So I will leave it here for now.
```
smooth_res_curve_approx <- function( b, midp, p ) {
    p_pow_b <- p ^ b
    f_p <- p_pow_b / ( midp ^ b + p_pow_b )
}
```
and
```
smooth_res_curve_approx_error <- function( b, midp, supply_curve ) {
    f_p <- smooth_res_curve_approx( b, midp, supply_curve$p )
    error <- f_p - supply_curve$percent_cumul
    crossprod( error, error )
}
```